components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java

```diff
@@ -120,6 +123,41 @@ public Optional<String> resolveResidentOrganization(String userId, String access
         return Optional.ofNullable(residentOrgId);
     }
 
+    @Override
+    public List<BasicOrganization> getHierarchyUptoResidentOrganization
+            (String userId, String accessedOrganizationId) throws OrganizationManagementException {
+
+        List<BasicOrganization> basicOrganizationList = new ArrayList<>();
+        try {
+            List<String> ancestorOrganizationIds =
+                    organizationManagementDAO.getAncestorOrganizationIds(accessedOrganizationId);
+            if (ancestorOrganizationIds != null) {
+                for (String organizationId : ancestorOrganizationIds) {
+                    String associatedTenantDomainForOrg = resolveTenantDomainForOrg(organizationId);
+                    if (StringUtils.isBlank(associatedTenantDomainForOrg)) {
+                        continue;
+                    }
+                    Optional<String> organizationName = organizationManagementDAO
+                            .getOrganizationNameById(organizationId);
+                    if (organizationName.isPresent()) {
+                        BasicOrganization basicOrganization = new BasicOrganization();
+                        basicOrganization.setId(organizationId);
+                        basicOrganization.setName(organizationName.get());
+                        basicOrganizationList.add(basicOrganization);
+                    }
+                    AbstractUserStoreManager userStoreManager = getUserStoreManager(associatedTenantDomainForOrg);
+                    if (userStoreManager.isExistingUserWithID(userId)) {
+                        break;
+                    }
+                }
+            }
+        } catch (UserStoreException e) {
+            throw handleServerException(ERROR_CODE_ERROR_WHILE_RESOLVING_ROOT_ORG, e, userId);
+        }
+        Collections.reverse(basicOrganizationList);
```

Comment:
Let's add a comment to mention the returning order.
Organizations will be sorted starting from resident organization(higher level) and ended up with the accessed organization(lower level)